src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
+     * @param array $files
```

Comment:
we need to define this in a doc comment. we probably should have a way to define a different file name even if we pass paths. what if that array was a hash map of filename => stream|absolute filename? or can a multipart contain the same filename multiple times?

if we allow both stream and path, the factory knowing about the implementation can convert as needed (open a stream from a file, resp. write the stream to a temporary file).